#160451597 reset password via email

Victor-Ugwueze opened the pull request:
#### What does this PR do?
- Reset password via email

#### Description of Task to be completed?
  - add AuthController.js
  - add ```resetPasseord``` and ```forgotPassword``` methods to AuthController
  - add authValidator.js to middleware to verify token from user's request and check the new password validity.
 - add route `/api/auth/forgot_password` and `/api/auth/reset_password`
  

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160451597](https://www.pivotaltracker.com/story/show/160451597 "reset password")
 
### Question
 NA

Victor-Ugwueze committed:
feat(reset passwod): reset password via email

 - add AuthController
   AuthController has two methods
    * resetPassword
    * forgotPassword

- add a validator to midleware/validator.js
  The validator checks the validity
  of the token

[Finishes #160451597]

KaiserPhemi commented on server/controllers/AuthController.js:
```diff
@@ -0,0 +1,80 @@
+            link,
+          });
+      })
+      .catch(e => e);
```
#### Critical
What's `e`? Be more explicit. 
`errors` or `error` is clearer

KaiserPhemi reviewed (changes requested):
#### Critical
- Implement feedback
- Fix merge conflict

Victor-Ugwueze commented:
I have fixed the conflict. I also changed package manager to yarn,  but I am having import module error.




<img width="1440" alt="screen shot 2018-10-09 at 1 07 30 am" src="https://user-images.githubusercontent.com/24890605/46639725-dcb54900-cb5f-11e8-95da-798d71f64ce5.png">

Victor-Ugwueze commented:
I have Implemented the feedback. Please review @KaiserPhemi, thank you.
